tests/scripts/all.sh

```diff
@@ -697,6 +697,31 @@ component_test_full_cmake_gcc_asan () {
     if_build_succeeded tests/compat.sh
 }
 
+component_test_zlib_make() {
+    msg "build: zlib enabled, make"
+    scripts/config.py set MBEDTLS_ZLIB_SUPPORT
+    make ZLIB=1 CFLAGS='-Werror -O1'
```

Comment:
This adds a build-time requirement that the development files for zlib must be available (`zlib.h` and `libz.so`). Should we assume that these files are always available? If so we should add a function similar to `check_tools` to check their availability. If not we should add a function `support_test_zlib_make` that checks their availability (and also `test_zlib_cmake`). Perhaps something like (untested):

```
support_test_zlib_make () {
    ret=0
    cat <<'EOF' >support_test_zlib_make_$$.c
#include "zlib.h"
int main(void) { return 0; }
EOF
    gcc -o support_test_zlib_make_$$.exe support_test_zlib_make_$$.c -lz || ret=$?
    rm -f support_test_zlib_make_$$.*
    return $ret
}
```
and
```
support_test_zlib_make () {
    support_test_zlib_cmake "$@"
}
```